src/chainstate/burn/operations/mod.rs

```diff
@@ -313,17 +314,32 @@ pub struct DelegateStxOp {
     pub burn_header_hash: BurnchainHeaderHash, // hash of the burn chain block header
 }
 
+fn hex_serialize<S: serde::Serializer>(bhh: &BurnchainHeaderHash, s: S) -> Result<S::Ok, S::Error> {
+    let inst = bhh.to_hex();
+    s.serialize_str(inst.as_str())
+}
+
+fn hex_deserialize<'de, D: serde::Deserializer<'de>>(
+    d: D,
+) -> Result<BurnchainHeaderHash, D::Error> {
+    let inst_str = String::deserialize(d)?;
+    BurnchainHeaderHash::from_hex(&inst_str).map_err(serde::de::Error::custom)
+}
+
 #[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
 pub struct PegInOp {
+    #[serde(serialize_with = "crate::chainstate::stacks::address::DisplaySerialize")]
+    #[serde(deserialize_with = "crate::chainstate::stacks::address::AddressDeser")]
     pub recipient: StacksAddress,
     pub recipient_contract_name: Option<String>, // If set, makes the recepient a smart contract principal
     pub peg_wallet_address: PoxAddress,
```

Comment:
Do you also want to add a serde helper to `peg_wallet_address` so it looks like a base58/bech32/bech32m address?

Reply:
It would be nicer, but I was avoiding having to write a `from_b58` function -- is this already implemented somewhere?

Reply:
It's a little janky right now, but you _could_ do this:

```rust
let addr_str = "<insert base58/bech32/bech32m string here>";
let pox_addr_opt = if let Some(btc_addr) = BitcoinAddress::from_string(addr_str) {
   PoxAddress::try_from_bitcoin_output(&BitcoinTxOutput { address: btc_addr, amount: u0 });
}
else {
   None
};

if let Some(pox_addr) = pox_addr_opt {
   /* do stuff with the PoX address */
}
else {
  /* error out */
}
```

Of course, I'd recommend packing the above into its own method in `PoxAddress` or some such, but the b58/bech32/bech32m parsing logic is already present and tested.

Reply:
Added in d4a8d3e